src/search-and-replace-dialog.jsx

```diff
@@ -0,0 +1,248 @@
+/** @jsx React.DOM */
+
+var React = require('react');
+var Util = require("./util.js");
+
+var SearchAndReplaceDialog = React.createClass({
+
+    getDefaultProps: function() {
+        return {
+            onChange() {}    // TODO: throw an error
+        };
+    },
+
+    getInitialState() {
+        return {
+            replaceString: "",
```

Comment:
```searchString``` and ```searchIndex``` are part of ```StatefulEditorPage```'s ```state``` because those values need to be passed down to each of the editors and their child components.  This feels like a job for a flux store, but since I haven't used flux before I'll need to do a little more investigation before I attempt this refactoring.